crate_universe/private/local_crate_mirror.bzl

```diff
@@ -0,0 +1,63 @@
+"""`local_crate_mirror` rule implementation.
+
+This is a private implementation detail of crate_universe, and should not be relied on in manually written code.
+
+This is effectively a `local_repository` rule impementation, but where the BUILD.bazel file is generated using the `cargo-bazel render` command.
+"""
+
+load("//crate_universe/private:common_utils.bzl", "execute")
+load("//crate_universe/private:generate_utils.bzl", "get_generator")
+load("//crate_universe/private:urls.bzl", "CARGO_BAZEL_SHA256S", "CARGO_BAZEL_URLS")
+load("//rust/platform:triple.bzl", "get_host_triple")
+
+def _local_crate_mirror_impl(repository_ctx):
+    path = repository_ctx.path(repository_ctx.attr.path)
+
+    host_triple = get_host_triple(repository_ctx)
+
+    generator, _generator_sha256 = get_generator(repository_ctx, host_triple.str)
+
+    execute(repository_ctx, ["bash", "-c", "cp -r {}/* {}/".format(path, repository_ctx.path("."))])
```

Comment:
Couldn't you symlink the paths here? If the contents of the files change it might help avoid rerunning repository rules but this one should be pretty quick either way.

Reply:
Originally I did, but _something_ didn't work, and I can't remember exactly what it was... It could be that `repository_ctx.read` doesn't track _through_ symlinks, just the symlink itself. Added a TODO with a comment to come back to.